src/sentry/tasks/relay.py

```diff
@@ -154,3 +108,159 @@ def schedule_update_config_cache(
+    schedule this task as that will take care of the queueing semantics.
+
+    The current implementation has some limitations:
+
+    - The task does not synchronise with the :func:`update_config_cache` task so depending
+      on when tasks complete they might have race conditions on writing the config to the
+      cache and an older version might be written.
+
+    - The task does not synchronise with more recent invocations of itself.
+
+    These will be addressed in the future using config revisions tracked in Redis.
+    """
+    validate_args(organization_id, project_id, public_key)
+
+    # Make sure we start by deleting out deduplication key so that new invalidation triggers
+    # can schedule a new message while we already started computing the project config.
+    projectconfig_debounce_cache.invalidation.mark_task_done(
+        organization_id=organization_id, project_id=project_id, public_key=public_key
+    )
+
+    keys = project_keys_to_update(
+        organization_id=organization_id, project_id=project_id, public_key=public_key
+    )
+
+    if organization_id:
+        # Previous incarnations of this task only delete all the affected configs in this
+        # case, relying of lazily filling them back in as they are requested.  Probably
+        # because some organizations can have thousands of projects and they may not all be
+        # active.  Do the same for now, but this could be improved.
+        projectconfig_cache.delete_many(key.public_key for key in keys)
+    else:
+        compute_project_configs(keys)
+
+
+def schedule_invalidate_config_cache(
+    *, trigger, organization_id=None, project_id=None, public_key=None
+):
+    """Schedules the :func:`invalidate_project_config` task.
+
+    This takes care of not scheduling a duplicate task if one is already scheduled.  The
+    parameters are passed straight to the task.
+    """
+    # TODO: so far i failed to merge this code with schedule_update_config_cache elegantly.
+    validate_args(organization_id, project_id, public_key)
+
+    if projectconfig_debounce_cache.invalidation.is_debounced(
+        public_key=public_key, project_id=project_id, organization_id=organization_id
+    ):
+        metrics.incr(
+            "relay.projectconfig_cache.skipped",
+            tags={"reason": "debounce", "update_reason": trigger, "task": "invalidation"},
+        )
+        # If this task is already in the queue, do not schedule another task.
+        return
+
+    metrics.incr(
+        "relay.projectconfig_cache.scheduled",
+        tags={"update_reason": trigger, "task": "invalidation"},
+    )
+    with sentry_sdk.push_scope() as scope:
+        scope.set_tag("update_reason", trigger)
+
+        invalidate_project_config.delay(
+            project_id=project_id,
+            organization_id=organization_id,
+            public_key=public_key,
+            update_reason=trigger,
+        )
```

Comment:
Does this even work across celery tasks?  I'm guessing this doesn't.

/cc @untitaker